### PR TITLE
Add PolyOverZq from Zq

### PR DESCRIPTION
**Pull request by MoogSven**
It is a common use case to create a Polynomial with just the constant coefficient. These can be used for calculations with other polynomials.

**Description**

This PR implements...
- [x] Add From<&Zq> for Zq
- [x] From<Into<Zq>> for PoyOverZq

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer_mod_q/poly_over_zq/from.rs [hunk 1]
```diff
@@ -172,6 +172,105 @@ impl FromStr for PolyOverZq {
     }
 }
 
+impl<IntoZq: Into<Zq>> From<IntoZq> for PolyOverZq {
```
Comment (Marvin-Beckmann) [inline src/integer_mod_q/poly_over_zq/from.rs:175]: IntoZq is kind of irritating to me and does not really fit to our naming (Integer, Rational, etc)

My suggestions:
- ResidueClass
- CongruenceClass
- IntegerModQ

File: src/integer_mod_q/poly_over_zq/from.rs [hunk 2]
```diff
@@ -172,6 +172,105 @@ impl FromStr for PolyOverZq {
...
+    /// ```
+    ///
+    /// # Panics ...
+    /// - if the provided value can not be converted into a [`Zq`].
+    ///   For example, if the modulus is not larger than one.
+    fn from(value: IntoZq) -> Self {
+        let value: Zq = value.into();
+        let poly_z = PolyOverZ::from(value.value);
+
+        PolyOverZq::from((&poly_z, &value.modulus))
```
Comment (Marvin-Beckmann) [inline src/integer_mod_q/poly_over_zq/from.rs:204]: provide ownership for the poly_z if it is later changed to a more efficient variant

**Review by Marvin-Beckmann — APPROVED** [review]

File: src/integer_mod_q/poly_over_zq/from.rs (hunk 1 above)
Comment (MoogSven) [reply src/integer_mod_q/poly_over_zq/from.rs:175]: Selected `IntegerModQ`

File: src/integer_mod_q/poly_over_zq/from.rs (hunk 2 above)
Comment (MoogSven) [reply src/integer_mod_q/poly_over_zq/from.rs:204]: PolyOverZq::from is not yet implemented for the ownership variant.
Therefore, I left it how it is at the moment.

**Pull request merged**